CMakeLists.txt

```diff
@@ -337,6 +337,26 @@ endif (WIN32)
 
 endif (OVERRIDE_CMAKE_CXX_FLAGS)
 
+OPTION(CMAKE_ENABLE_CODE_COVERAGE "Enable code coverage" OFF)
+
+if(CMAKE_ENABLE_CODE_COVERAGE)
+
+  string(TOUPPER ${CMAKE_BUILD_TYPE} UPPERCASE_CMAKE_BUILD_TYPE)
+  if(NOT UPPERCASE_CMAKE_BUILD_TYPE STREQUAL DEBUG)
+    message( WARNING "Code coverage results with an optimised (non-Debug) build may be misleading" )
+  endif(NOT UPPERCASE_CMAKE_BUILD_TYPE STREQUAL DEBUG)
+
+  if(CLR_CMAKE_PLATFORM_UNIX)
+    add_definitions(-fprofile-arcs -ftest-coverage -fPIC)
```

Comment:
@janvorli I get this error when linking shared library libsos.so. Here is a more verbose output

*Linking CXX shared library libsos.so
cd /home/sridhar/coreclr/bin/obj/Linux.x64.Debug/src/ToolBox/SOS/Strike && /usr/bin/cmake -E cmake_link_script CMakeFiles/sos.dir/link.txt --verbose=1
/usr/bin/clang++-3.5  -fPIC  -Wall -Wno-null-conversion -std=c++11 -g -O0   --coverage -shared -Wl,-soname,libsos.so -o libsos.so CMakeFiles/sos.dir/disasm.cpp.o CMakeFiles/sos.dir/datatarget.cpp.o CMakeFiles/sos.dir/eeheap.cpp.o CMakeFiles/sos.dir/exts.cpp.o CMakeFiles/sos.dir/gcroot.cpp.o CMakeFiles/sos.dir/metadata.cpp.o CMakeFiles/sos.dir/sildasm.cpp.o CMakeFiles/sos.dir/stressLogDump.cpp.o CMakeFiles/sos.dir/strike.cpp.o CMakeFiles/sos.dir/sos.cpp.o CMakeFiles/sos.dir/util.cpp.o CMakeFiles/sos.dir/disasmX86.cpp.o ../../../inc/libcorguids.a ../../../debug/shim/libdebugshim.a ../../../debug/dbgutil/libdbgutil.a ../../../dlls/mscordac/libmscordaccore.so ../../../inc/libcorguids.a ../../../debug/dbgutil/libdbgutil.a -Wl,--start-group ../../../vm/dac/libcee_dac.a ../../../debug/ee/dac/libcordbee_dac.a ../../../debug/daccess/libdaccess.a ../../../md/compiler/dac/libmdcompiler_dac.a ../../../md/hotdata/dac/libmdhotdata_dac.a ../../../md/runtime/dac/libmdruntime_dac.a ../../../md/enc/dac/libmdruntimerw_dac.a ../../../strongname/api/dac/libstrongname_dac.a ../../../utilcode/dac/libutilcode_dac.a ../../../unwinder/dac/libunwinder_dac.a -Wl,--end-group ../../../dlls/mscorrc/full/libmscorrc_debug.a ../../../pal/src/libcoreclrpal.a -lpthread -lrt -ldl -lunwind ../../../palrt/libpalrt.a -Wl,-rpath,/home/sridhar/coreclr/bin/obj/Linux.x64.Debug/src/dlls/mscordac 
/usr/bin/ld: ../../../inc/libcorguids.a(cordebug_i.c.o): relocation R_X86_64_PC32 against undefined symbol `llvm_gcda_start_file' can not be used when making a shared object; recompile with -fPIC
/usr/bin/ld: final link failed: Bad value
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [src/ToolBox/SOS/Strike/libsos.so] Error 1
make[2]: Leaving directory `/home/sridhar/coreclr/bin/obj/Linux.x64.Debug'
make[1]: *** [src/ToolBox/SOS/Strike/CMakeFiles/sos.dir/all] Error 2
make[1]: /usr/bin/cmake -E cmake_progress_report /home/sridhar/coreclr/bin/obj/Linux.x64.Debug/CMakeFiles *